plugins/plugin-chart-echarts/src/Gauge/controlPanel.tsx

```diff
@@ -0,0 +1,267 @@
+  valueFormatter,
+  valueAnimation,
+  startAngle,
+  endAngle,
+  intervals,
+  intervalColorIndices,
+} = DEFAULT_FORM_DATA;
+
+const config: ControlPanelConfig = {
+  controlPanelSections: [
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        ['groupby'],
+        ['metric'],
+        ['adhoc_filters'],
+        ['row_limit', null],
+        [
+          {
+            name: 'min_val',
+            config: {
+              ...sharedControls.metric,
+              type: 'MetricsControl',
+              label: t('Min'),
+              description: t('Minimum value on the gauge axis'),
+              default: minVal,
+              validators: [validateNonEmpty],
+            },
+          },
+        ],
+        [
+          {
+            name: 'max_val',
+            config: {
+              ...sharedControls.metric,
+              type: 'MetricsControl',
+              label: t('Max'),
+              description: t('Maximum value on the gauge axis'),
+              default: maxVal,
+              validators: [validateNonEmpty],
+            },
+          },
+        ],
+      ],
+    },
+    {
+      label: t('Chart Options'),
+      expanded: true,
+      controlSetRows: [
+        [<h1 className="section-header">{t('General')}</h1>],
+        [
+          {
+            name: 'title',
+            config: {
+              type: 'TextControl',
+              label: t('Title'),
+              renderTrigger: true,
+              default: title,
+              description: t('Chart title'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'number_format',
+            config: {
+              type: 'SelectControl',
+              freeForm: true,
+              label: t('Number format'),
+              renderTrigger: true,
+              default: numberFormat,
+              choices: D3_FORMAT_OPTIONS,
+              description: `${t('D3 format syntax: https://github.com/d3/d3-format')} ${t(
+                'Only applies when "Label Type" is set to show values.',
+              )}`,
+            },
+          },
+        ],
+        [
+          {
+            name: 'font_size',
+            config: {
+              type: 'SliderControl',
+              label: t('Font size'),
+              renderTrigger: true,
+              min: 10,
+              max: 25,
+              default: fontSize,
+              description: t('Font size for chart elements'),
+            },
+          },
+        ],
+        [<h1 className="section-header">{t('Progress Bar')}</h1>],
+        [
+          {
+            name: 'roundcap',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Roundcap'),
+              renderTrigger: true,
+              default: roundcap,
+              description: t('Style the ends of the progress bar with a roundcap'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'show_split_line',
```

Comment:
These names need to be in lower camel case.

Reply:
Sorry, I saw `adhoc_filters` and `row_limit`, so I assumed the custom ones should follow.

Reply:
oh my bad, this is correct.